# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/abort/AbortSignal.h
# Commit: d27c32ae0908
# Full Hash: d27c32ae0908c0b5d2c9213b9a624e3d0b71e2d5
# Author: Vincent Hilla <vhilla@mozilla.com>
# Date: 2024-01-23 21:57:50
# Regressor Bug: 1830781
# File Overlap Count: 1
# Description:
#   Bug 1830781 - Implement AbortSignal.any() r=saschanaz,webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D198011
# ==============================================================================

diff -r 8af7503ecb57 -r d27c32ae0908 dom/abort/AbortSignal.h
--- a/dom/abort/AbortSignal.h	Tue Jan 23 08:04:52 2024 +0000
+++ b/dom/abort/AbortSignal.h	Tue Jan 23 08:43:07 2024 +0000
@@ -7,6 +7,7 @@
 #ifndef mozilla_dom_AbortSignal_h
 #define mozilla_dom_AbortSignal_h
 
+#include "mozilla/RefPtr.h"
 #include "mozilla/dom/AbortFollower.h"
 #include "mozilla/DOMEventTargetHelper.h"
 
@@ -47,6 +48,10 @@
                                                uint64_t aMilliseconds,
                                                ErrorResult& aRv);
 
+  static already_AddRefed<AbortSignal> Any(
+      GlobalObject& aGlobal,
+      const Sequence<OwningNonNull<AbortSignal>>& aSignals);
+
   void ThrowIfAborted(JSContext* aCx, ErrorResult& aRv);
 
   // AbortSignalImpl
@@ -57,8 +62,17 @@
 
   virtual bool IsTaskSignal() const { return false; }
 
+  bool Dependent() const;
+
  protected:
   ~AbortSignal();
+
+  void MakeDependentOn(AbortSignal* aSignal);
+
+  nsTArray<WeakPtr<AbortSignal>> mSourceSignals;
+  nsTArray<RefPtr<AbortSignal>> mDependentSignals;
+
+  bool mDependent;
 };
 
 }  // namespace mozilla::dom